src/compiler/types.ts

```diff
@@ -4310,8 +4310,10 @@ namespace ts {
         root: ConditionalRoot;
         checkType: Type;
         extendsType: Type;
-        resolvedTrueType?: Type;
-        resolvedFalseType?: Type;
+        resolvedTrueType: Type;
+        resolvedFalseType: Type;
+        /* @internal */
+        inferredTrueType?: Type; // The `trueType` instantiated with the `combinedMapper`, if present

```

Comment:
I suggest renaming to `trueType`, `falseType`, and `resolvedInferredTrueType`. Elsewhere we use the "resolvedXXX" pattern to indicate a lazily computed entity.